torch/_inductor/codegen/triton.py

```diff
@@ -1073,16 +1097,24 @@ def print_min(expr):
                         return f"min({texpr(expr[0])}, {print_min(expr[1:])})"
 
                 size = print_min(sizes)
-            # The conditions need to be in parens because of Python's operator precedence.
-            # It'd be less # error-prone to use and/or/not, which is suported by triton
-            cond = f"((0 <= {var}) & ({var} < {size}))"
-            cond_print = f"0 <= {var} < {size}"
-            if not isinstance(original_index, sympy.Integer):
-                var_mask = f"({mask})" if "&" in mask else mask
-                var_mask = f" | ~{var_mask}"
+            gen_min, gen_max = gen_min_max(var, size)
+            # FooBar
```

Comment:
stray comment ?

Reply:
I was thinking for a minute whether I could simplify this code as it has some repetition, but then I realized that this is its own version of the classic coding interview question "FooBar" lol